mesonbuild/mintro.py

```diff
@@ -265,19 +237,27 @@ def func_project(self, node, args, kwargs):
         options = {k: v for k, v in self.environment.cmd_line_options.items() if k.startswith('backend_')}
 
         self.coredata.set_options(options)
-        self.add_languages(proj_langs)
+        self.func_add_languages(None, proj_langs, None)
 
     def do_subproject(self, dirname):
         subproject_dir_abs = os.path.join(self.environment.get_source_dir(), self.subproject_dir)
         subpr = os.path.join(subproject_dir_abs, dirname)
         try:
-            subi = BuildoptionsInterperter(subpr, '', self.backend, cross_file=self.cross_file, subproject=dirname, subproject_dir=self.subproject_dir, env=self.environment)
+            subi = IntrospectionInterpreter(subpr, '', self.backend, cross_file=self.cross_file, subproject=dirname, subproject_dir=self.subproject_dir, env=self.environment)
             subi.analyze()
+            subi.project_data['name'] = dirname
+            self.project_data['subprojects'] += [subi.project_data]
         except:
             return
 
     def func_add_languages(self, node, args, kwargs):
-        return self.add_languages(self.flatten_args(args))
+        args = self.flatten_args(args)
+        need_cross_compiler = self.environment.is_cross_build() and self.environment.cross_info.need_cross_compiler()
```

Comment:
__[Flake8]__ [F841] local variable 'need_cross_compiler' is assigned to but never used

<sub>posted by Sider</sub>